packages/nextjs-template/components/DendronSearch.tsx

```diff
@@ -0,0 +1,80 @@
+import { LoadingStatus } from "@dendronhq/common-frontend";
+import { AutoComplete, Alert } from "antd";
+import React from "react";
+import { useCombinedDispatch } from "../features";
+import { browserEngineSlice } from "../features/engine";
+import { useFetchFuse } from "../utils/fuse";
+import {
+  DendronCommonProps,
+  DendronPageWithNoteDataProps,
+  verifyNoteData,
+} from "../utils/types";
+import DendronSpinner from "./DendronSpinner";
+
+const LOADING_KEY = "__loading";
+const LOADING_MESSAGE = "Loading...";
+
+export function DendronSearch(props: DendronCommonProps) {
+  if (!verifyNoteData(props)) {
+    return <DendronSpinner />;
+  }
+
+  return <DendronSearchComponent {...props} />;
+}
+
+function DendronSearchComponent(props: DendronPageWithNoteDataProps) {
+  const dispatch = useCombinedDispatch();
+  const { noteIndex, dendronRouter } = props;
+  const [value, setValue] = React.useState("");
+  const { ensureIndexReady, fuse, error, loading } = useFetchFuse(props.notes);
+  if (error) {
+    return (
+      <Alert
+        type="error"
+        closable={false}
+        message="Error loading data for the search."
+      />
+    );
+  }
+  return (
+    <>
+      <AutoComplete
+        onClick={ensureIndexReady}
+        style={{ width: "100%" }}
+        value={value}
+        onChange={setValue}
+        onSelect={(_selection, option) => {
+          const id = option.key?.toString()!;
+          dendronRouter.changeActiveNote(id, { noteIndex });
+          dispatch(
+            browserEngineSlice.actions.setLoadingStatus(LoadingStatus.PENDING)
+          );
+          setValue("");
+        }}
+        placeholder="Search"
+      >
+        {loading ? (
+          <AutoComplete.Option
+            key={LOADING_KEY}
+            value={LOADING_MESSAGE}
+            disabled
+          >
+            <div>{LOADING_MESSAGE}</div>
+          </AutoComplete.Option>
+        ) : undefined}
+        {fuse
+          ? fuse.search(value).map((note) => (
+              <AutoComplete.Option key={note.item.id} value={note.item.fname}>
+                <div>
+                  {note.item.title}
+                  <span style={{ marginLeft: "10px", opacity: 0.7 }}>
```

Comment:
can we add a preview for the fusejs results? so when the results are rendered, show the raw text for the search result

```
{10 chars context} {{textThatIsMatched}} {10 chars context}
```

Reply:
fuse.js isn't giving me any info for which part of the document is matched. Do you know if there's a setting that I might be missing for that?